pkg/server/tunnel.go

```diff
@@ -144,5 +144,18 @@ func (t *Tunnel) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			"connectionID", connection.connectID)
 	}
 
+	packet := &client.Packet{
```

Comment:
If we're going to do this it would make sense to have the added code as a separate (anonymous) deferred function after the deferred close call on the defer stack.

Reply:
+1

Reply:
@cheftako ok, add a anonymous deferred function to include the added code and the deferred conn close.